homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -19,7 +24,30 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up DSMR Reader sensors."""
+    """Set up DSMR Reader sensors via configuration.yaml and show deprecation warning."""
+    _LOGGER.warning(
+        "DSMR Reader yaml config is now deprecated and has been imported. "
+        "Please remove it from your config"
+    )
+    current_entries = hass.config_entries.async_entries(domain=DOMAIN)
+    if not current_entries:
+        await hass.config_entries.async_add(
+            ConfigEntry(
+                version=1,
+                domain=DOMAIN,
+                title="DSMR Reader",
+                data={},
+                source=SOURCE_PLATFORM_CONFIG,
+            )
+        )
```

Comment:
Please implement async_step_import instead

Reply:
I tried to, but the method didn't trigger at any point. I never really figured out why it didn't work, but I presume it has something to do with there being no configuration other mentioning the dmsr_reader as a sensor. If you're convinced it should be able to work with `async_step_import` as well, I'm willing to give it another shot.

Reply:
You need to start it here. On mobile so I can't put a suggestion but you can 'git grep SOURCE_IMPORT' to find an example

Reply:
Ah, the fact that I need to trigger it didn't register with me. I've now got a working version locally that does this properly. It also fixed the situation where the integration was imported AND suggested at the same time, so that's great. I'll first check whether it has had any impact on the tests and will then commit.

Reply:
Just wanted to confirm that the requested changes have been made. I can't remember whether resolving a conversation actually sends out a notification.